versioned/storage/mongodb/src/main/java/org/projectnessie/versioned/storage/mongodb/MongoDBPersist.java

```diff
@@ -271,6 +265,15 @@ public Reference markReferenceAsDeleted(@Nonnull @jakarta.annotation.Nonnull Ref
     return reference.withDeleted(true);
   }
 
+  private Bson referenceCondition(Reference reference) {
+    List<Bson> filters = new ArrayList<>(5);
```

Comment:
```suggestion
    List<Bson> filters = new ArrayList<>(3);
```

Reply:
That number will be correct in the future ;)